dipy/workflows/tracking.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+from __future__ import division
+
+import logging
+import numpy as np
+
+import nibabel as nib
+from nibabel.streamlines import save, Tractogram
+
+from dipy.data import get_sphere
+from dipy.direction import DeterministicMaximumDirectionGetter, PeaksAndMetrics
+from dipy.io.image import load_nifti
+from dipy.io.peaks import load_peaks
+from dipy.tracking import utils
+from dipy.tracking.local import (ThresholdTissueClassifier,
+                                 LocalTracking)
+from dipy.workflows.workflow import Workflow
+
+
+class GenericTrackFlow(Workflow):
+    def _core_run(self, stopping_path, stopping_thr, seeding_path, seed_density,
+                  use_sh, pam, out_tract):
+
+        stop, affine = load_nifti(stopping_path)
+        classifier = ThresholdTissueClassifier(stop, stopping_thr)
+        logging.info('classifier done')
+        seed_mask, _ = load_nifti(seeding_path)
+        seeds = \
+            utils.seeds_from_mask(
+                seed_mask,
+                density=[seed_density, seed_density, seed_density],
+                affine=affine)
+        logging.info('seeds done')
+        direction_getter = pam
+
+        if use_sh:
+            direction_getter = \
+                DeterministicMaximumDirectionGetter.from_shcoeff(
+                    pam.shm_coeff,
+                    max_angle=30.,
+                    sphere=pam.sphere)
+
+        #seeds = seeds[:50000, :]
+        streamlines = LocalTracking(direction_getter, classifier,
+                                    seeds, affine, step_size=.5)
+        logging.info('LocalTracking initiated')
+       
+        tractogram = Tractogram(streamlines, affine_to_rasmm=np.eye(4))
+        save(tractogram, out_tract)
+
+        logging.info('Saved {0}'.format(out_tract))
+
+
+class DetTrackPAMFlow(GenericTrackFlow):
+    @classmethod
+    def get_short_name(cls):
+        return 'tracking'
+
+    def run(self, pam_files, stopping_files, seeding_files,
+            stopping_thr=0.2,
+            seed_density=1,
+            use_sh=False,
+            out_dir='',
+            out_tractogram='tractogram.trk'):
+
+        """ Workflow for deterministic using a saved peaks and metrics (PAM)
+        file as input.
+
+        Parameters
+        ----------
+        pam_files : string
+           Path to the peaks and metrics files. This path may contain
+           wildcards to use multiple masks at once.
+        stopping_files : string
+            Path of FA or other images used for stopping criteria for tracking.
+        seeding_files : string
+            A binary image showing where we need to seed for tracking.
+        stopping_thr : float, optional
+            Threshold applied to stopping volume's data to identify where
+            tracking has to stop. (default 0.25)
+        seed_density : int, optional
```

Comment:
This is the number of equidistant points in each dimension of the voxel.
For example seed_density of 2 means 2x2x2 = 8 points per voxel
This can be problematic for some applications because just moving from 1 to 2 will mean 8 times more streamlines. Bago used this because he was trying to make absolutely sure that the points are distributed in an equal way around the voxel.

Reply:
Was this the best way to do this? Probably not. We need to figure out a way were the seed density will actually mean number of points in the voxel and not the current definition. But let's do this in another PR. This is not a stopper for here.

Reply:
That's fine -- I don't think we should change the way it works (definitely not on this PR). But it would be good to provide a bit more detail in the documentation about what it does and what the units are (I understand it's not exactly a unitful quantity). Maybe write the example you provided in your answer here in the docstring?

Reply:
Sure.